src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs

```diff
@@ -99,21 +100,17 @@ public static bool TryFormat(float value, Span<byte> destination, out int bytesW
                 utf16Text = value.ToString(new string(formatText), CultureInfo.InvariantCulture);
             }
 
-            // Copy the value to the destination, if it's large enough.
+            bytesWritten = 0;
 
-            if (utf16Text.Length > destination.Length)
```

Comment:
Why remove this check?

Reply:
I would have expected that `Utf8Encoding.GetBytes` would have done this check (going from UTF16 to UTF8 will, at best, have the same number of characters, and at worst will have more; but I don't think it will ever have less), but it looks like it doesn't....

Reply:
Fixed. But, unless I'm missing something, I think we should consider removing this and adding the check to `Utf8Encoding.GetBytes` instead.

Reply:
I'm not sure why we didn't make it a Try method. But as we didn't, what would it do as is? This is also in Corelib, so it could use internals. And we could if course as additional public APIs.

Reply:
Hmmm, that's a good point. `Utf8Encoding.GetBytes` lets you do partial writes, but we don't in the Utf8Formatter.

Reply:
Actually, it looks like `Utf8Encoding.GetBytes` will always throw if the destination isn't big enough: https://source.dot.net/#System.Private.CoreLib/shared/System/Text/UTF8Encoding.cs,1045.

So we could just updadte it to fast path that throw, maybe also provide a `Try` overload that returns 0 rather than throwing.